apps/wakustealthcommitments/stealth_commitment_protocol.nim

```diff
@@ -0,0 +1,76 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  ../../waku/common/logging,
+  ../../waku/waku_node,
+  ../../waku/waku_rln_relay,
+  ./node_spec,
+  ./wire_spec
+
+export 
+  wire_spec,
+  logging
+
+type StealthCommitmentProtocol* = object
+  wakuApp: App
+  contentTopic: string
+
+
+proc SCPHandler*(msg: WakuMessage) =
+  let decodedRes = WakuStealthCommitmentMsg.decode(msg.payload)
+  if decodedRes.isErr():
+    error "could not decode scp message"
+  let decoded = decodedRes.get()
+  ## do something with the decoded message
+  info "received SCP message: ", decoded
+
+
+proc new*(wakuApp: App, contentTopic = ContentTopic("/wakustealthcommitments/1/app/proto")): StealthCommitmentProtocol =
+  let SCP = StealthCommitmentProtocol(wakuApp: wakuApp, contentTopic: contentTopic)
+
+  proc handler(topic: PubsubTopic, msg: WakuMessage): Future[void] {.async, gcsafe.} =
+    if msg.contentTopic == contentTopic:
+      SCPHandler(msg)
+
+  wakuApp.node.subscribe((kind: PubsubSub, topic: DefaultPubsubTopic), some(handler))
```

Comment:
hey @alrevuelta, for some reason while running this application, this handler gets called for messages it is sending. am i missing a config field?

Reply:
noticed we set `triggerSelf` to `true` here -> https://github.com/waku-org/nwaku/blob/731dfcbdf67998c11660b6997d811a55af21c5b2/waku/waku_relay/protocol.nim#L166

Reply:
yep